### ISSUE TITLE
syntax highlighting

### ISSUE THREAD

**Issue by ducin** [raml2html/raml2html#178]
Hello,

First of all - congratulations on this tool - job well done! I especially like that it's so... small - and at the same time - so useful!

One of the things I'm missing in raml2html is that embedding either `example`s or `schema`s is really unreadable, when I've got a narrow column and no syntax highlighting. I'm using `raml2html` in a big enterprise project and I've got big JSON documents and readability counts in this matter.

The solution I came up with is quite simple:
* use syntax highlighter - but not a server-side one, but client-side
* I found there is a JavaScript tool that does exactly this job: https://github.com/isagalaev/highlight.js
* moreover, there's no need to embed it to the output of `raml2html`. [cdn.js](https://cdnjs.com/libraries/highlight.js/) hosts the library and it could be simply accessed by from HTML script tag.
* and if the peer doesn't have internet connection (so that cdn.js would not work), well, sorry :) the other solution is to include highlight.js into the output, but I find it a worse solution.

So what do you think about such feature?

**Comment by fireantology** [comment]
Hi,
I think you can already accomplish that by simply using custom templates instead of built in ones.

**Comment by ducin** [comment]
@fireantology didn't know that, thank you. But it requires extra work from everyone. Why not adding it into the core?

**Comment by fireantology** [comment]
Hi @ducin,
I can't talk for the maintainer, but in my experience you will easily end up in customizing the template anyway because everyone wants something in a different way.
Off course is always possible to improve the defaults templates but probably is better to give priority to more essential features.
You can always contribute :)

Regards

**Comment by ducin** [comment]
Getting back to this one :)

I know there might be more essential features in such library, but... it takes significantly more time to read the entire codebase in order to solve critical issues. I think at this step I'd stick to something smaller and add syntax highlighting (using external js highlighting lib) - I'm using raml2html anyway and I'd find it useful. When I have big schemas and example JSONs, it's damn hard to read through un-highlighted JSONs. In the future, perhaps, I'd contribute to more critical ones :)

But before I create the PR, I'd like to know if the maintainer(s?) would accept such feature (or reject it). So *how about adding syntax highlighting - do you want me to add it into raml2html default template*?

**Comment by mikestowe** [comment]
I don't see why there would be any issue with adding syntax highlighting to the core template.  Let's call in the files remotely (unless there are any objections to this).  

If you need any help on this just let me know :)

**Comment by ducin** [comment]
@mikestowe That was an easy one :) a true one liner :D it was just to replace:
`$('.page-header pre code, .top-resource-description pre code')`...
with
`$('.page-header pre code, .top-resource-description pre code, .modal-body pre code')`...

The `.page-header pre code` stands for:
![image](https://cloud.githubusercontent.com/assets/375027/14788219/48860708-0b07-11e6-9dee-68736344c802.png)

`.top-resource-description pre code` stands for:
![image](https://cloud.githubusercontent.com/assets/375027/14788250/626dca16-0b07-11e6-9264-cc5bfbe3a7c4.png)

but the examples and schemas inside the popups were missing:
![image](https://cloud.githubusercontent.com/assets/375027/14788274/7c589b36-0b07-11e6-90f5-3260d84a9ed4.png)

and have now been added with `.modal-body pre code`:
![image](https://cloud.githubusercontent.com/assets/375027/14788315/9f04222c-0b07-11e6-9225-2d1e6add807f.png)

**Comment by aspectmkn8rd** [comment]
I would love to see JSON highlight in core template.

**Comment by ducin** [comment]
@aspectmkn8rd that has been already done in #197 - see `raml2html:develop`.
@mikestowe can we release either `2.5.0` or `2.4.1` with the feature? I can help/assist if it's needed.

**Comment by aspectmkn8rd** [comment]
@ducin ah ok. But if I understand this correctly: right now there is no easy way for me to achieve this? (I'm using raml2html via gulp plugin.)

**Comment by ducin** [comment]
I hope @mikestowe will allow me to release or will do it himself ;) - and you will have nothing non-standard to do. But for now, you can `git clone` raml2html locally and don't use the gulp plugin (I guess you're using `gulp-raml2html` which is being updated right now by me :) ).

**Comment by mikestowe** [comment]
There are two pull requests pending that I would like to review before releasing (just to get everything in).  But once those are reviewed and everything's tested again for quality purposes I'm happy to get this released :)  Sorry for the delay, should just be one or two days more if everyone's onboard.

Thanks!!!

**Comment by aspectmkn8rd** [comment]
@mikestowe ok thank you. Looking forward for it.

**Comment by aspectmkn8rd** [comment]
so is there like any updates?..

**Comment by mkn8rd** [comment]
![tw](https://cloud.githubusercontent.com/assets/855706/15647587/2d11035c-266d-11e6-88b7-8dfd372d1cf5.gif)

**Comment by ducin** [comment]
Dear `raml2html` Team,

I have submitted this piece 3 months ago, since then, it has been waiting on [`develop` branch](https://github.com/raml2html/raml2html/tree/develop). I've been waiting for this feature to be available in an official release (available via standard `npm install raml2html`)... and waiting, and waiting for ages. And I feel like this:

![waiting](https://cloud.githubusercontent.com/assets/375027/17077993/f07e4752-50e6-11e6-9124-dcae049241e4.jpg)

Can you please publish raml2html `2.5.0` or `2.4.1`, whatever, including current `develop` content? I like RAML very much, but please, try to be more active! Thanks in advance :) :beers:

**Comment by sichvoge** [comment]
@ducin can you assist with reviewing open PRs? Let's do that first and if everything is ok please feel free to release. What do you think?

**Comment by sichvoge** [comment]
BTW, I am so sorry that we were not responsive so far. Shame on!

**Comment by kevinrenskers** [comment]
2.5.0 was just released and includes this fix.
